### DISCUSSION TITLE
スライドショーの再生順は固定ですか？

### DISCUSSION THREAD

**Discussion by HanadaNico** [torum/Image-viewer#31] [Q&A] [answered]
現在batファイルでスライドショーを開始するところまでは行けたのですが、
スライドショーはランダム再生offのとき、
名前順からのみ再生されるのでしょうか。

（もしかしたら連続投稿になっているかもしれませんすみません。）

**Comment by torum** [comment, marked as answer]
こんばんは、コメントありがとうございます。

再生順は基本的にはOS（Windows上ではWindows）依存で、OSが返してくるファイル一覧の順になっています。今の所、再生順を任意に変更できるようにはなっていません（ランダム化以外には）。

実はこれ、ちょうど数日前にIssueの方(#27)で、アルファベット順にソートできないか、というのを頂いていました。そこで今思ったのですが、アルファベット順だけでなく、更新日時や作成日時などの順にソートするオプションを作ると良いかなぁ、と。

ただ、この「アルファベット順」というのが非常に厄介で、数字が混じった時どうするか、とか、色々ありまして・・・文字列の比較でもロジカルソートとナチュラルソートとかで順番が全然違って来たり・・・・しかも日本語が混じったらどうするか、サブフォルダが絡んだ場合は、など、考えていたらちょっと簡単ではなかったので、すぐには出来ないなぁという所であります。

**Reply by HanadaNico** [reply]
早速の返信ありがとうございます。

検索かけたときチラッと「大昔からのOSの仕様」と見かけましたが、OSの仕様なんですね・・・。

スライドの時間調整が容易にできるソフトも少ないようなので、
今後も重宝させていただきます。
また、陰ながら応援させていただきます。

回答ありがとうございました！

**Comment by HanadaNico** [comment]
9割型Copilotのコードコピペしただけですが、
添付したtxt内のフォルダ名を変更し、txtからbatに変更することで、
実質名前を変更せずに日時順でスライドショーされるように変更しました。

logfileが出力されますが気にしないでください、。

やっていること
1.スライドショーを行いたい画像ファイルが存在するフォルダーを取得
2.フォルダー内のファイル名を取得、保存
3.ファイル名を日時順に変更
4.ImageViewer起動、スライドショー開始
5.ImageViewerを終了
6.変更されたファイル名を元のファイル名に変更

やっていることは単純ですが、bat初心者には十分難問でした。

[featCopilot.txt](https://github.com/user-attachments/files/17095551/featCopilot.txt)

**Comment by torum** [comment]
なるほど！そういう方法があったかと自分の想像外の方法で、勉強になりました。
